CMSgov/easi-app
# EASI-2445 - Send email notifications when feedback is given on a TRB request

# EASI-2445

## Changes and Description

- Send email notifications when feedback is provided on a TRB request, both for requesting edits and approving request as ready for consult.
- Create a separate template for the common EASi header and refactor existing templates to use it.
- Add TRB emails to `cmd/test_email_templates/main.go` for easy manual testing.

## How to test this change

1. Spin the backend up, open the GraphQL playground, and open Mailcatcher on localhost:1080.
2. Create a TRB request, noting the returned ID for later. GQL:
```gql
mutation {
  createTRBRequest(requestType: NEED_HELP) {
    id
    type
  }
}
```

3. Create and submit a form for the request, using the request ID from above. GQL:
```
mutation createAndSubmitForm {
  updateTRBRequestForm(
    input: { isSubmitted: true, trbRequestId: "trbRequestIdFromAbove" }
  ) {
    status
    trbRequestId
  }
}
```

4a. Request edits on the submitted form, using the request ID from above. The EUA IDs should come from the list of mock users in `pkg/local/cedar_ldap.go`. GQL:
```gql
mutation requestEdits {
  createTRBRequestFeedback(
    input: {
      trbRequestId: "trbRequestIdFromAbove"
      feedbackMessage: "I am disappoint"
      copyTrbMailbox: false
      notifyEuaIds: ["ABCD", "GRTB"]
      action: REQUEST_EDITS
    }
  ) {
    id
    trbRequestId
    feedbackMessage
    copyTrbMailbox
    notifyEuaIds
    action
  }
}
```
Mailcatcher should show the email notification requesting edits to the emails; the addressees will be the email addresses associated with the IDs in `notifyEuaIds`; if `copyTrbMailbox` is also selected, it will also include `TRB_EMAIL`.

4b. With a fresh request, provide feedback that the request is ready for consult. As with requesting edits, the EUA IDs should come from the list of mock users in `pkg/local/cedar_ldap.go`. GQL:
```gql
mutation readyForConsult {
  createTRBRequestFeedback(
    input: {
      trbRequestId: "trbRequestIdFromAbove"
      feedbackMessage: "Looks good!"
      copyTrbMailbox: false
      notifyEuaIds: ["CMSU"]
      action: READY_FOR_CONSULT
    }
  ) {
    id
    trbRequestId
    feedbackMessage
    copyTrbMailbox
    notifyEuaIds
    action
  }
}
```
Mailcatcher should show the email notification that the request is ready for the consult meeting, using the same logic for addressees as when requesting edits.

## Notes
* The email templates contain links to both the task list view (which work) and the admin view (which don't). The links to the admin view will be fixed in EASI-2488, once we have the URL for that view.
* `pkg/email/templates/easi_header.gohtml` intentionally lacks an ending newline so it can be added to other templates without adding unexpected spacing.

## Screenshots

Email requesting edits:
![trb feedback notification - edits requested - multiple links](https://user-images.githubusercontent.com/92891039/208992104-d1558ad8-e22c-4b81-846d-5d4bcb17388b.PNG)


Email saying request is ready for consult:
![trb feedback notification - ready for consult - multiple links](https://user-images.githubusercontent.com/92891039/208992094-50b12695-13fd-4b8c-addb-9d634d7ff1ae.PNG)
